phoenicis-javafx/src/main/java/org/phoenicis/javafx/controller/library/LibraryController.java

```diff
@@ -68,6 +78,48 @@ private void showError(Exception e) {
         });
     }
 
+    /**
+     * sets the default category icons for the library sidebar buttons
+     *
+     * We cannot use the default category buttons for this purpose for several reasons:
+     * - It cannot be guaranteed that the repository which contained the category of a certain app does still exist
+     * after the app has been installed (e.g. the repository might have been deleted).
+     * - It should not be required to fetch the repository to use the library.
+     *
+     * Therefore, use the following approach:
+     * - When installing an app, store the icon for its category together with the shortcut (i.e. in the library).
+     * - Use this icon as fallback if the theme does not specify an icon for this category.
+     *
+     * @param categories list of shortcut categories
+     */
+    private void setDefaultLibraryCategoryIcons(List<ShortcutCategoryDTO> categories) {
```

Comment:
This method seems to be quite similar to `setDefaultCategoryIcons` above. Can we maybe merge both methods into one single method?

Reply:
Technically yes, the question is where it should be located.

Reply:
I think we should either add an abstract parent class or an interface containing the method.

Reply:
An interface may be better because a parent class should be extended by all feature panel controllers

Reply:
I don't like this. In my opinion, this is not some functionality that is implemented by the controllers so it should not be an interface nor an abstract class. Logically it should be in the `ThemeManager` but I don't want to have the DTOs there.
I would suggest to leave it as it is right now.